http_parser.c

```diff
@@ -1782,9 +1782,9 @@ size_t http_parser_execute (http_parser *parser,
 
         if (parser->flags & F_TRAILING) {
           /* End of a chunked request */
-          UPDATE_STATE(NEW_MESSAGE());
-          CALLBACK_NOTIFY(message_complete);
-          break;
+          UPDATE_STATE(s_message_done);
```

Comment:
Sure, I asked him (by mail) to respond to some of your questions earlier this morning, he'll hopefully chime in soon. In the mean time I tried an alternative implementation that;s slightly closer to your original one:
```
         UPDATE_STATE(s_message_done);
          CALLBACK_NOTIFY_NOADVANCE(chunk_complete);
          UPDATE_STATE(NEW_MESSAGE());
          CALLBACK_NOTIFY(message_complete);
          break;
```
This variant passes tests and just adds the 2 lines for firing the chunk_complete event, so the diff may look slightly less scary. I have no strong preference for either solution, just slightly prefer the shorter committed one that reuses the 'case s_message_done:' (fewer LOC, plus we had this variant in production in the Proxygen branch for a long time).